modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/SearchPermissionCheckerImpl.java

```diff
@@ -622,10 +599,10 @@ private String _getPermissionName(
 	private static final Log _log = LogFactoryUtil.getLog(
 		SearchPermissionCheckerImpl.class);
 
-	private final Collection<SearchPermissionFieldContributor>
-		_searchPermissionFieldContributors = new CopyOnWriteArrayList<>();
-	private final Collection<SearchPermissionFilterContributor>
-		_searchPermissionFilterContributors = new CopyOnWriteArrayList<>();
+	private volatile ServiceTrackerList<SearchPermissionFieldContributor>
```

Comment:
@W2jx Why it needs to be volatile?

Reply:
Hi @tinatian , SF added it automatically.